nvtabular/io/parquet.py

```diff
@@ -677,23 +744,101 @@ def _close_writers(self):
+        super().__init__(out_dir, **kwargs)
+        self.md_collectors = {}
+        self.pwriter_kwargs = {"compression": None}
+
+    @property
+    def _pwriter(self):
+        return pwriter_pyarrow
+
+    def _read_parquet(self, source):
+        return pd.read_parquet(source, engine="pyarrow")
+
+    def _get_row_group_size(self, df):
+        # Make sure our `row_group_size` argument (which corresponds
+        # to the number of rows in each row-group) will produce
+        # row-groups ~128MB in size.
+        if not hasattr(self, "_row_group_size"):
+            row_size = df.memory_usage(deep=True).sum() / max(len(df), 1)
+            self._row_group_size = math.ceil(128_000_000 / row_size)
+        return self._row_group_size
+
+    def _to_parquet(self, df, sink):
+        md = []
+        df.to_parquet(
+            sink,
+            row_group_size=self._get_row_group_size(df),
+            metadata_collector=md,
+            compression=None,
+            index=False,
+        )
+        fn = sink.split(self.fs.sep)[-1]
+        md[0].set_file_path(fn)
+        return md
+
+    def _append_writer(self, path, schema=None):
+
+        # Define "metadata collector" for pyarrow
+        _md_collector = []
+        _args = [schema]
+        _kwargs = {"metadata_collector": _md_collector}
+
+        # Use `BaseParquetWriter` logic
+        super()._append_writer(path, add_args=_args, add_kwargs=_kwargs)
+
+        # Keep track of "metadata collector" for pyarrow
+        self.md_collectors[path] = _md_collector
+
+    def _write_table(self, idx, data):
+        table = pa.Table.from_pandas(data)
+        writer = self._get_or_create_writer(idx, schema=table.schema)
+        writer.write_table(table, row_group_size=self._get_row_group_size(data))
+
+    @classmethod
+    def write_special_metadata(cls, md, fs, out_dir):
+        # Sort metadata by file name and convert list of
+        # tuples to a list of metadata byte-blobs
+        md_list = [m[1] for m in sorted(list(md.items()), key=lambda x: natural_sort_key(x[0]))]
+
+        # Aggregate metadata and write _metadata file
+        _write_pq_metadata_file_pyarrow(md_list, fs, out_dir)
 
-def _write_pq_metadata_file(md_list, fs, path):
+    def _close_writers(self):
+        for writer, path in zip(self.data_writers, self.data_paths):
+            fn = path.split(self.fs.sep)[-1]
+            writer.close()
+            self.md_collectors[path][0].set_file_path(fn)
+        return self.md_collectors
+
+
+def _write_pq_metadata_file_cudf(md_list, fs, path):
     """ Converts list of parquet metadata objects into a single shared _metadata file. """
     if md_list:
         metadata_path = fs.sep.join([path, "_metadata"])
         _meta = cudf.io.merge_parquet_filemetadata(md_list) if len(md_list) > 1 else md_list[0]
+        # import pdb; pdb.set_trace()
         with fs.open(metadata_path, "wb") as fil:
-            _meta.tofile(fil)
+            # _meta.tofile(fil)
```

Comment:
```suggestion
```

Reply:
Oops - Sorry!  (forgot I was testing this while reviewing Alberto's PR)